File: backend/app/schemas/gallery.py
from typing import Optional
from pydantic import BaseModel, ConfigDict
import datetime


class GalleryPhotoBase(BaseModel):
    url: str
    alt: Optional[str] = None
    label: str
    caption: Optional[str] = None
    event_name: Optional[str] = None
    category: Optional[str] = "Workshop"
    date: Optional[str] = None
    featured: Optional[bool] = False
    order: Optional[int] = 1


class GalleryPhotoCreate(GalleryPhotoBase):
    pass


class GalleryPhotoUpdate(BaseModel):
    url: Optional[str] = None
    alt: Optional[str] = None
    label: Optional[str] = None
    caption: Optional[str] = None
    event_name: Optional[str] = None
    category: Optional[str] = None
    date: Optional[str] = None
    featured: Optional[bool] = None
    order: Optional[int] = None


class GalleryPhotoResponse(GalleryPhotoBase):
    id: str
    created_at: Optional[datetime.datetime] = None
    updated_at: Optional[datetime.datetime] = None

    model_config = ConfigDict(from_attributes=True)

File: src/api/gallery.ts
import { apiClient } from "./client"

export interface GalleryPhoto {
  id: string
  url: string
  alt?: string
  label: string
  caption?: string
  eventName?: string
  event_name?: string
  category: "Workshop" | "Hackathon" | "Symposium" | "Campus" | "Seminar" | "Other"
  date?: string
  featured?: boolean
  order?: number
  created_at?: string
}

export const galleryApi = {
  getAll: () => apiClient<GalleryPhoto[]>("/api/gallery"),
  create: (data: Partial<GalleryPhoto>) =>
    apiClient<GalleryPhoto>("/api/gallery", {
      method: "POST",
      body: JSON.stringify(data),
      requireAuth: true,
    }),
  update: (id: string, data: Partial<GalleryPhoto>) =>
    apiClient<GalleryPhoto>(`/api/gallery/${id}`, {
      method: "PUT",
      body: JSON.stringify(data),
      requireAuth: true,
    }),
  delete: (id: string) =>
    apiClient<{ message: string; id: string }>(`/api/gallery/${id}`, {
      method: "DELETE",
      requireAuth: true,
    }),
}
